library/Asana/Api/Task.hs

```diff
@@ -71,10 +65,13 @@ instance ToJSON CustomFields where
   toJSON (CustomFields fields) = object $ concatMap toPair fields
    where
     toPair = \case
-      CustomNumber gid _ n -> [gidToText gid .= n]
-      e@(CustomEnum gid _ _ _) -> [gidToText gid .= customEnumId e]
+      CustomNumber gid _ n -> [gidToKey gid .= n]
+      e@(CustomEnum gid _ _ _) -> [gidToKey gid .= customEnumId e]
       _ -> []
 
+    -- fromString will give us Text for aeson-1.x and Key for aeson-2.x
+    gidToKey = fromString . T.unpack . gidToText
```

Comment:
Hopefully GHC can optimize away this `fromString . unpack` in the `Text` scenario, but the extra allocations aren't a huge hurt.